easy_my_coop_taxshelter_report/models/tax_shelter_declaration.py

```diff
@@ -62,11 +62,11 @@ class TaxShelterDeclaration(models.Model):
 
     def _excluded_from_declaration(self, entry):
         if entry.date >= self.date_from and entry.date <= self.date_to:
-            declaration = self
+            declarations = self
         else:
-            declaration = self.search([('date_from', '<=', entry.date),
+            declarations = self.search([('date_from', '<=', entry.date),
                                        ('date_to', '>=', entry.date)])

```

Comment:
Same comment as Manuel, I don't understand this `if else`.

Reply:
In fact, we need to know of the cooperator was excluded or not from the tax shelter. So we need to get the declaration corresponding to the subscription date of the shares aka entry.date

It return the current declaration if the shares have subscribed during the declaration year or the corresponding declaration (in the past).